servers/jax-rs-tests/src/main/java/org/projectnessie/jaxrs/gc/AbstractRestGCTest.java

```diff
@@ -0,0 +1,427 @@
+        true);
+  }
+
+  @Test
+  public void testSingleRefRenameTable() throws BaseNessieClientServerException {
+    String prefix = "singleRefRenameTable";
+    Map<String, Set<Content>> expectedExpired = new HashMap<>();
+
+    Branch branch1 = createBranch(prefix);
+    // one commit for "table_2" on branch1
+    CommitOutput table2 =
+        commitSingleOp(
+            prefix, branch1, branch1.getHash(), 42, "cid_2", "table_2", "file1", null, null);
+    // expect this commit to be expired as it is before cutoff time.
+    fillExpectedContents(branch1, 1, expectedExpired);
+
+    final Instant cutoffTime = Instant.now();
+
+    // two commit for "table_2" on branch1
+    table2 =
+        commitSingleOp(
+            prefix, branch1, table2.hash, 43, "cid_2", "table_2", "file2", table2.content, null);
+    table2 =
+        commitSingleOp(
+            prefix, branch1, table2.hash, 44, "cid_2", "table_2", "file3", table2.content, null);
+
+    // rename table "table_2" to "table_2_renamed" on branch1
+    table2 =
+        commitSingleOp(
+            prefix,
+            branch1,
+            table2.hash,
+            44,
+            "cid_2",
+            "table_2_renamed",
+            "file3",
+            table2.content,
+            "table_2");
+
+    // rename table should not expire the live commits after cutoff timestamp.
+    performGc(
+        cutoffTime,
+        null,
+        expectedExpired,
+        new HashSet<>(Arrays.asList(prefix + "_cid_1", prefix + "_cid_2")),
+        true);
+  }
+
+  @Test
+  public void testMultiRefDropRef() throws BaseNessieClientServerException {
+    String prefix = "multiRefDropRef";
+    Map<String, Set<Content>> expectedExpired = new HashMap<>();
+
+    Branch branch1 = createBranch(prefix + "_1");
+    Branch branch2 = createBranch(prefix + "_2");
+    // commit for "table_1" on branch1
+    CommitOutput table1 =
+        commitSingleOp(
+            prefix, branch1, branch1.getHash(), 42, "cid_1", "table_1", "file1", null, null);
+    // expect this commit to be expired as it is before cutoff time.
+    fillExpectedContents(branch1, 1, expectedExpired);
+    // commit for "table_2" on branch2
+    CommitOutput table2 =
+        commitSingleOp(
+            prefix, branch2, branch2.getHash(), 42, "cid_2", "table_2", "file1", null, null);
+    // expect this commit to be expired as it is before cutoff time.
+    fillExpectedContents(branch2, 1, expectedExpired);
+
+    final Instant cutoffTime = Instant.now();
+
+    // commits for "table_1" on branch 1
+    table1 =
+        commitSingleOp(
+            prefix, branch1, table1.hash, 44, "cid_1", "table_1", "file3", table1.content, null);
+
+    // commits for "table_2" on branch2
+    table2 =
+        commitSingleOp(
+            prefix, branch2, table2.hash, 44, "cid_2", "table_2", "file3", table2.content, null);
+    // expect this commit to be expired as branch2 is dropped below and there is no other reference
```

Comment:
This sounds legit and correct. What I'm not sure about is whether user shall have the ability to "undo a drop of a branch/tag", which would only work until GC ran.
OTOH this should be protected via the "new commit protection", right?